src/promptflow/promptflow/_sdk/_serving/flow_invoker.py

```diff
@@ -201,3 +201,38 @@ def _dump_invoke_result(self, invoke_result):
                 invoke_result.output, base_dir=self._dump_to, sub_dir=Path(".promptflow/output")
             )
             dump_flow_result(flow_folder=self._dump_to, flow_result=invoke_result, prefix=self._dump_file_prefix)
+
+
+class AsyncFlowInvoker(FlowInvoker):
+    async def _invoke_async(self, data: dict, run_id=None, disable_input_output_logging=False):
+        log_data = "<REDACTED>" if disable_input_output_logging else data
+        self.logger.info(f"Validating flow input with data {log_data!r}")
+        validate_request_data(self.flow, data)
+        self.logger.info(f"Execute flow with data {log_data!r}")
+        # Pass index 0 as extension require for dumped result.
+        # TODO: Remove this index after extension remove this requirement.
```

Comment:
I guess we don't need index=0 here? This is only for extension + pf test

Reply:
let me double check